pkgs/by-name/zi/zipline/package.nix

```diff
@@ -29,111 +28,81 @@ in
-    pname = "yarn-deps";
-    inherit (finalAttrs) version src patches;
-
-    nativeBuildInputs = [ yarn-berry ];
-
-    dontInstall = true;
-
-    YARN_ENABLE_TELEMETRY = "0";
-    NODE_EXTRA_CA_CERTS = "${cacert}/etc/ssl/certs/ca-bundle.crt";
-
-    configurePhase = ''
-      export HOME="$NIX_BUILD_TOP"
-      yarn config set enableGlobalCache false
-      yarn config set cacheFolder $out
-      yarn config set --json supportedArchitectures.os '[ "linux" ]'
-      yarn config set --json supportedArchitectures.cpu '[ "arm64", "x64" ]'
-    '';
-
-    buildPhase = ''
-      mkdir -p $out
-      yarn install --immutable --mode skip-build
-    '';
-
-    outputHash = "sha256-niO+obo1JHAoWbLgjf1ttB6UqTCCjEuhiILvfT3O0q4=";
-    outputHashMode = "recursive";
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-rDm3LFFB65SdSfqABMZelhfx4Cq6u0EV3xdDp9lBR54=";
   };
 
+  buildInputs = [ vips ];
+
   nativeBuildInputs = [
-    pkg-config
-    makeWrapper
+    pnpm_9.configHook
     nodejs
-    yarn-berry
-    python3
-  ];
-
-  buildInputs = [
-    vips
-    openssl
+    makeWrapper
   ];
 
-  env = {
-    YARN_ENABLE_TELEMETRY = "0";
-    ZIPLINE_DOCKER_BUILD = "true";
-  } // environment;
-
-  configurePhase = ''
-    export HOME="$NIX_BUILD_TOP"
-    yarn config set enableGlobalCache false
-    yarn config set cacheFolder $yarnOfflineCache
-  '';
+  env = environment;
 
   buildPhase = ''
-    mkdir -p $HOME/.node-gyp/${nodejs.version}
-    echo 9 > $HOME/.node-gyp/${nodejs.version}/installVersion
-    ln -sfv ${nodejs}/include $HOME/.node-gyp/${nodejs.version}
-    export npm_config_nodedir=${nodejs}
+    runHook preBuild
+
+    pnpm build
 
-    yarn install --immutable --immutable-cache
-    yarn build
+    runHook postBuild
   '';
 
   installPhase = ''
+    runHook preInstall
+
     mkdir -p $out/{bin,share/zipline}
 
-    cp -r dist public node_modules .next prisma next.config.js mimes.json package.json $out/share/zipline
+    cp -r build node_modules prisma .next mimes.json code.json package.json $out/share/zipline
 
     mkBin() {
       makeWrapper ${lib.getExe nodejs} "$out/bin/$1" \
         --chdir "$out/share/zipline" \
+        --set NODE_ENV production \
         --prefix PATH : ${lib.makeBinPath [ openssl ]} \
         --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ openssl ]} \
         ${
           lib.concatStringsSep " " (
             lib.mapAttrsToList (name: value: "--set ${name} ${lib.escapeShellArg value}") environment
           )
         } \
-        --add-flags "--enable-source-maps $2"
+        --add-flags "--enable-source-maps build/$2"
     }
 
-    mkBin zipline dist
-    for script in dist/scripts/*.js; do
-      mkBin "zipline-$(basename $script .js)" "$script"
+    mkBin zipline server
+    mkBin ziplinectl ctl
+
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    find $out -name libvips-cpp.so.42 -print0 | while read -d $'\0' libvips; do
```

Comment:
Is this one hardcoded location? Then we could just write that and notice immeadetely when it breaks with the next update

Reply:
There are at least two (see https://github.com/Defelo/nixpkgs-review-gha/actions/runs/13439969913/job/37551655506#step:6:2113), I don't know if hardcoding makes sense here

Reply:
ufff, yeah, we keep it like that for now